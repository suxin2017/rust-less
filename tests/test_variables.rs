#[cfg(feature = "serde")]
use lightningcss::stylesheet::{ParserOptions, StyleSheet};

#[cfg(feature = "serde")]
macro_rules! test_ast {
  ($code:expr) => {
    let stylesheet = StyleSheet::parse($code, ParserOptions::default()).unwrap();
    insta::assert_yaml_snapshot!(stylesheet);
  };
}

#[cfg(feature = "serde")]
#[test]
fn quick_test() {
  test_ast!(
    r#"
@a: 123;
.a {
  @a: 456;

  .b {
    @c: f;
  }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_media() {
  test_ast!(
    r#"
@a: 123;
@media screen and (min-width: 900px) {
  @b: 456;
  article {
    @c: 789;
    padding: 1rem 3rem;
  }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_keyframes() {
  test_ast!(
    r#"
@a: 123;
@keyframes slidein {
    @b: 456;
    from {
        @c: 789;
        transform: translateX(0%);
    }

    to {
        transform: translateX(100%);
    }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_font_face() {
  test_ast!(
    r#"
@a: 123;
@font-face {
    @b: 456;
  font-family: "MyHelvetica";
  src: local("Helvetica Neue Bold"), local("HelveticaNeue-Bold"),
    url("MgOpenModernaBold.ttf");
  font-weight: bold;
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_font_palette_values() {
  test_ast!(
    r#"
@a: 123;
@font-palette-values --identifier {
@b: 123;
font-family: Bixa;
}
  "#
  );
}
#[cfg(feature = "serde")]
#[test]
fn test_at_rule_page() {
  test_ast!(
    r#"
@a: 123;
@page {
  @b: 123;
}
@page {
  @c: 123;
  @top-right {
    @d: 123;
  }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_supports() {
  test_ast!(
    r#"
@a: 123;
@supports (display: flex) {
  @b: 123;
  .flex-container {
    @c: 123;
    display: flex;
  }
}
  "#
  );
}
#[cfg(feature = "serde")]
#[test]
fn test_at_counter_style() {
  test_ast!(
    r#"
    @a: 123;
@counter-style thumbs {
@b:456;
  system: cyclic;
  symbols: "\1F44D";
  suffix: " ";
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_nest_rule() {
  test_ast!(
    r#"
@a: 123;
.a {
  @a: 456;

  .b {
    @c: f;
  }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_layer() {
  test_ast!(
    r#"
@layer utilities {
  @a: 123;
}

@layer utilities {
  @a: 123;
  p {
    @b: 123;
  }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_property() {
  test_ast!(
    r#"
    @a:123;
@property --item-size {
@b:345;
syntax: "<percentage>";
inherits: true;
initial-value: 40%;
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_container() {
  test_ast!(
    r#"
    @a:123;
@container (width > 400px) {
@b: 123;
  h2 {
  @c: 345;
    font-size: 1.5em;
  }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_scope() {
  test_ast!(
    r#"
    @a:123;
@scope (.article-body) to (figure) {
 @b: 123;
  h2 {
  @c: 345;
    font-size: 1.5em;
  }
}
  "#
  );
}

#[cfg(feature = "serde")]
#[test]
fn test_at_rule_starting_style() {
  test_ast!(
    r#"
    @a:123;
@starting-style {
 @b:123;
  [popover]:popover-open {
   @c:123;
    opacity: 0;
    transform: scaleX(0);
  }
}
  "#
  );
}
