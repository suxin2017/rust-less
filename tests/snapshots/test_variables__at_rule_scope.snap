---
source: tests/test_variables.rs
expression: stylesheet
---
rules:
  - type: scope
    value:
      scopeStart:
        - - type: class
            name: article-body
      scopeEnd:
        - - type: type
            name: figure
      rules:
        - type: style
          value:
            selectors:
              - - type: type
                  name: h2
            declarations:
              importantDeclarations: []
              declarations:
                - property: font-size
                  value:
                    type: length
                    value:
                      type: dimension
                      value:
                        unit: em
                        value: 1.5
              variables: []
            rules: []
            loc:
              source_index: 0
              line: 4
              column: 3
            variables:
              - name: c
                value:
                  - type: token
                    value:
                      type: number
                      value: 345
      loc:
        source_index: 0
        line: 2
        column: 1
      variables:
        - name: b
          value:
            - type: token
              value:
                type: number
                value: 123
variables:
  - name: a
    value:
      - type: token
        value:
          type: number
          value: 123
sources:
  - ""
sourceMapUrls:
  - ~
licenseComments: []

